src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -288,6 +296,60 @@ export class EdgeProvider extends Bridgeable {
     Actions.pop()
   }
 
+  async getTransactions () {
+    // Get Wallet Info
+
+    const guiWallet = UI_SELECTORS.getSelectedWallet(this._state)
+    const coreWallet = CORE_SELECTORS.getWallet(this._state, guiWallet.id)
+    const currencyCode = UI_SELECTORS.getSelectedCurrencyCode(this._state)
+
+    // Prompt user with yes/no modal for permission
+
+    const confirmTxShare = await Airship.show(bridge => (
+      <TwoButtonSimpleConfirmationModal
+        bridge={bridge}
+        title={s.strings.fragment_wallets_export_transactions}
+        subTitle={sprintf(s.strings.transaction_history_permission, coreWallet.name)}
+        cancelText={s.strings.no}
+        doneText={s.strings.yes}
+      />
+    ))
+
+    // Grab transactions from current wallet
+
+    if (confirmTxShare) {
+      const fiatCurrencyCode = coreWallet.fiatCurrencyCode
+      const balance = coreWallet.getBalance({ currencyCode })
+
+      const txs = await coreWallet.getTransactions({ currencyCode })
+      const transactions: Array<EdgeTransaction> = []
+      for (const tx of txs) {
+        const newTx: EdgeTransaction = {
+          currencyCode: tx.currencyCode,
+          nativeAmount: tx.nativeAmount,
+          networkFee: tx.networkFee,
+          parentNetworkFee: tx.parentNetworkFee,
+          blockHeight: tx.blockHeight,
+          date: tx.date,
+          signedTx: '',
+          txid: tx.txid,
+          ourReceiveAddresses: tx.ourReceiveAddresses,
+          metadata: tx.metadata
+        }
+        transactions.push(newTx)
+      }
+      const result: EdgeGetTransactionsResult = {
+        fiatCurrencyCode,
+        balance,
+        transactions
```

Comment:
We do we include this extra stuff in the reply? Don't they already have this via other API's? If not, maybe we should name `getTransactions` tp `getWalletHistory` instead, since it does more than it says on the tin.